rust/kernel/chrdev.rs

```diff
@@ -149,12 +197,9 @@ unsafe impl<const N: usize> Sync for Registration<{ N }> {}
 impl<const N: usize> Drop for Registration<{ N }> {
     fn drop(&mut self) {
         if let Some(inner) = self.inner.as_mut() {
-            // SAFETY: Calling unsafe functions, `0..inner.used` of
-            // `inner.cdevs` are initialized in `Registration::register`.
+            // SAFETY: TODO needs explanation why unregister_chrdev_region
```

Comment:
I think something that says that `self.inner` being `Some(_)` means that a previous call to `alloc_chrdev_region` completed successfully would be fine here.